inc/osvr/Client/RenderManagerConfig.h

```diff
@@ -0,0 +1,104 @@
+/** @file
+    @brief OSVR rendermanager configuration
+
+    @date 2015
+
+    @author
+    Sensics, Inc.
+    <http://sensics.com>
+
+    */
+
+// Copyright 2015 Sensics, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef INCLUDED_RenderManagerConfig_h_GUID_C8DA5781_5B6C_454A_B4FF_1DB50CBE3479
+#define INCLUDED_RenderManagerConfig_h_GUID_C8DA5781_5B6C_454A_B4FF_1DB50CBE3479
+
+// Required for DLL linkage on Windows
+#include <osvr/Client/Export.h>
+#include <osvr/Util/UniquePtr.h>
+#include <osvr/Util/ClientOpaqueTypesC.h>
+
+// Standard includes
+#include <string>
+#include <iostream>
+#include <exception>
+#include <vector>
+
+namespace osvr {
+namespace client {
+
+    class RenderManagerConfig;
+    typedef unique_ptr<RenderManagerConfig> RenderManagerConfigPtr;
+    class RenderManagerConfigFactory {
+    public:
+        OSVR_CLIENT_EXPORT static RenderManagerConfigPtr
+            create(OSVR_ClientContext ctx);
+    };
+
+    class RenderManagerConfig {
+    public:
+        OSVR_CLIENT_EXPORT RenderManagerConfig(const std::string& render_manager_config);
+        OSVR_CLIENT_EXPORT RenderManagerConfig() {};
+
+        void OSVR_CLIENT_EXPORT parse(const std::string& render_manager_config);
+        void OSVR_CLIENT_EXPORT print() const;
+
+        /// Read the property information.
+        bool OSVR_CLIENT_EXPORT getDirectMode() const;
+        unsigned OSVR_CLIENT_EXPORT getDisplayIndex() const;
```

Comment:
Preferable to either use an explicit typedef (if it's comparable to something in the public API) to an explicitly-sized int, or an explicitly-sized int. The exception is `std::size_t` which is OK (suggested, even, when that's what it's describing - it's a nice meaningful name) except in public API (where the fact that it varies in size between platforms frequently, vs somewhat-less-frequently for the other un-sized int types, make it a pain to FFI)